R/generate_pt_nowcast_mat_list.R

```diff
@@ -64,16 +64,60 @@ generate_pt_nowcast_mat_list <- function(reporting_triangle_list,
   } else { # create a list with the same pmf
     delay_pmf_list <- rep(list(delay_pmf), length(reporting_triangle_list))
   }
-  # Iterate through each item in both lists of reporting triangles
-  # and delay PMFs
+
+  safe_generate_pt_nowcast_mat <- .safelydoesit(generate_pt_nowcast_mat)
+
+  # Use the safe version in mapply, iterating through each item in both
+  # lists of reporting triangles and delay PMFs
   pt_nowcast_mat_list <- mapply(
-    generate_pt_nowcast_mat,
+    function(triangle, pmf, ind) {
+      result <- safe_generate_pt_nowcast_mat(
+        reporting_triangle = triangle,
+        delay_pmf = pmf,
+        n = n,
+        max_delay = max_delay
+      )
+      if (!is.null(result$error)) {
+        # Print the index and the error message
+        error_msg <- sprintf(
+          "Error at index %d: Point nowcast matrix could not be generated - %s",
+          ind,
+          result$error$message
+        )
+        message(error_msg)
+        return(NULL)
+      } else {
+        # Return the result if successful
+        return(result$result)
+      }
+    },
     reporting_triangle_list,
     delay_pmf_list,
-    MoreArgs = list(max_delay = max_delay, n = n),
+    seq_along(reporting_triangle_list),
     SIMPLIFY = FALSE
   )
 
+  # After running, filter the results to find error indices
+  error_indices <- which(sapply(pt_nowcast_mat_list, is.null))
+  # Print summary
+  if (length(error_indices) == length(reporting_triangle_list)) {
```

Comment:
In terms of a user experience it would probably better if there was the summary of the errors followed by the detail vs the detail followed by a summary. I think it is fine to move forward like this but we may want a low priority issue to update this in the future.

Reply:
Yeah this prints within the iterator, in a similar way to stan where it tells you whats happening as its happening and then gives a summary. I prefer the errors being printed in order of the code running but not a strong preference. Will make an issue.